test/parallel/test-console-table.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const { Console } = require('console');
+
+const queue = [];
+
+const console = new Console({ write: (x) => {
+  queue.push(x);
+}, removeListener: () => {} }, process.stderr, false);
+
+function test(x, y, z) {
+  let args = [x];
+  let expected = y;
+  if (arguments.length === 3) {
+    args = [x, y];
+    expected = z;
+  }
+  console.table(...args);
+  assert.strictEqual(queue.shift(), expected);
+}
+
+/* eslint-disable no-tabs */
+
+test({ a: 1 }, `(index)	Values
+a	1
+`);
+
+test([{ a: 1 }, { b: 2 }], `(index)	a	b
+0	1	undefined
+1	undefined	2
```

Comment:
I think it would be best to add extra padding up to the value length that has the highest length of the column. That way it will improve the readability massively.

Reply:
i was considering that but i decided that as this isn't really intended to do more than humor the spec, and in general tables in terminals have a history of being formatted with single tabs without compensation (especially in configs and stuff) so i wanted to stay consistent with that.

Reply:
I feel like it improves the readability a lot, so I would really like to add that. I do not see any issues about inconsistencies here.

Reply:
Showing the value `undefined` is actually wrong here out of my perspective because there is just no value. So it should be empty instead of showing undefined. Otherwise this can not be distinguished from the value `undefined`.

As an extra test I would add:

```js
test(
  [{ a: 1 }, { b: undefined }], 
  '(index) a b' +
  '0       1 ' +
  '1         undefined'
)
```